### PR TITLE
feat: add user profile bloc

### PR DESCRIPTION
**Pull request by dtscalac**
# Description

Adds dummy user profile management

## Screenshots

Screenshot 2024-09-05 at 09 08 52

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

**Comment by digitalheartxs** [comment]
Nice implementation 🙌🎉
However, in our case, do we really need bloc over cubit?

**Comment by dtscalac** [comment]
> Nice implementation 🙌🎉 However, in our case, do we really need bloc over cubit?

In the future the bloc will have many more events to update / edit profile, etc. Question is, does using bloc hurt us in any way that cubit solves this pain? Bloc has more capabilities and slightly more complex syntax because of it. Imho, we should stick to one, not mix them. I.e. in the future we might need a centralized logging solution for blocs/cubits. Wouldn't like to implement it separately for cubit and separately for blocs.

File: catalyst_voices/uikit_example/lib/main.dart [hunk 1]
```diff
@@ -27,6 +28,7 @@ class _UIKitExampleAppState extends State<UIKitExampleApp> {
     return MultiBlocProvider(
       providers: [
         BlocProvider.value(value: _sessionBloc),
+        BlocProvider.value(value: _userProfileBloc),
```
Comment (damian-molinski) [inline catalyst_voices/uikit_example/lib/main.dart:31]: Should we add it in Voices `App`->`MultiBlocProvider`? As well as register in `Dependencies`

Comment (dtscalac) [reply catalyst_voices/uikit_example/lib/main.dart:31]: Will do. Wasn't sure when we will move to work on cat-voices app instead of UI kit

Comment (damian-molinski) [reply catalyst_voices/uikit_example/lib/main.dart:31]: Soon, after my PR i think :D

Comment (dtscalac) [reply catalyst_voices/uikit_example/lib/main.dart:31]: Ok. Is there anything that is blocking this PR from merging then? Once you're finished I will do the changes

**Review by damian-molinski — APPROVED** [review]

**Pull request merged**